### PR TITLE
Add category hook demo

### PR DESCRIPTION
**Pull request by msharafjr**

### CODE REVIEW

**Review by shadyvb — CHANGES_REQUESTED** [review]
Categories are arrays, and they have a slug and a name.

File: src/app/store/hooks/usePost.tsx [hunk 1]
```diff
@@ -12,8 +14,8 @@ export const usePost = ( id: number ): Post => {
         meta: {
             _thumbnail_id: 2,
         },
-        category: 1,
-        author: 1,
+        categories: [useCategory(1)],
```
Comment (shadyvb) [inline src/app/store/hooks/usePost.tsx]: Spacing here is not correct, make sure spacing is automatically fixed in your editor. It should happen automatically if you have TSLint installed in vscode.

File: src/app/store/types/post.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import {Category} from './category';
```
Comment (shadyvb) [inline src/app/store/types/post.ts]: Spacing issues here as well.

**Review by shadyvb — CHANGES_REQUESTED** [review]

**Comment by msharafjr** [comment]
@shadyvb

File: src/app/store/types/post.ts [hunk 3]
```diff
@@ -7,6 +11,8 @@ export type Post = {
         rendered: string,
     },
     meta: {
-        _thumbnail_id: number,
+        _thumbnail: Attachment,
```
Comment (shadyvb) [inline src/app/store/types/post.ts]: It is `_thumbnail_id`, that's the API definition

**Review by shadyvb — CHANGES_REQUESTED** [review]

**Comment by msharafjr** [comment]
@shadyvb

**Review by shadyvb — APPROVED** [review]

**Pull request merged**